src/client.ts

```diff
@@ -53,6 +53,10 @@ export class DelphinusContract {
     let pastEvents = [];
     let start = fromBlock;
     let end = 0;
+    if(fromBlock > toBlock){
+      console.log("No New Blocks Found From:"+ fromBlock);
+      return {"events": [], "breakpoint": fromBlock}
```

Comment:
Is this correct?
Assume it is just normal scenario and latest block number not increased.
For example, lastCheckedBlock = 10, and reliableBlockNumber=10
So we didn't check 11,
However by this code, it will return 11 and await db.updatelastCheckedBlockNumber(pastEvents.breakpoint); will update last checked block to 11.

Reply:
I think adjust the breakpoint value is not a good solution and easy to have bug.
It should do not call API and return null.

And at the place which call it, add code to treat null.
Never use any logic like -1 as it is not maintainable and scalable. (Think about if in the future we change the from from +1 to other number, it is very hard to figure out and fix the code logic.